core/server/utils/image-size.js

```diff
@@ -0,0 +1,60 @@
+var sizeOf = require('image-size'),
+    url = require('url'),
+    Promise = require('bluebird'),
+    http = require('http'),
+    https = require('https'),
+    dimensions,
+    request,
+    prot;
+
+module.exports.getImageSize = function getImageSize(imagePath, timeout) {
+    return new Promise(function imageSizeRequest(resolve) {
+        var timer,
+            imageObject = {},
+            timerEnded = false,
+            options = url.parse(imagePath);
+
+        imageObject.url = imagePath;
+        prot = imagePath.indexOf('https') === 0 ? https : http;
+        options.headers = {'User-Agent': 'Mozilla/5.0'};
+
+        request = prot.get(options, function (res) {
+            var chunks = [];
+            clearTimeout(timer);
+
+            res.on('data', function (chunk) {
+                chunks.push(chunk);
+            });
+
+            res.on('end', function () {
+                if (res.statusCode === 200 && !timerEnded) {
+                    try {
+                        dimensions = sizeOf(Buffer.concat(chunks));
+
+                        imageObject.width = dimensions.width;
+                        imageObject.height = dimensions.height;
+
+                        return resolve(imageObject);
+                    } catch (err) {
```

Comment:
Any recommendation, how to deal with a thrown error here, or would it be fine the way I did it now, which just returns the url for the image?